### PR TITLE
[RLlib; Tune] Fix WandB metric overlap after restore from checkpoint.

### PR DESCRIPTION
**Pull request by sven1977**
Fix WandB metric overlap after restore from checkpoint.

This PR fixes the following problem:
* An Algorithm is running through `Tuner.fit()`, checkpoints every 100 iterations, and uses the WandB integration.
* Let's assume the algo (trial) crashes at iteration 154 and then the user performs a `Tuner.resume()` to resume it from the last checkpoint (which would have happened at iteration 100).
* However, WandB has already logged all the results from iterations 101 to 154. So when restoring, the WandB logger should make sure that the results from the (repeated) 101 to 154 are re-sent and properly logged on the WandB side.

To fix this, we'll make sure to always send the explicit `step` along with each `wandb.log()` call.

Note that upon restoring from a checkpoint, RLlib already properly sets `Trainable.training_iteration` to the value stored in the last checkpoint. Other `Trainable` subclasses might have to implement a similar logic in their checkpointing behavior or we'll eventually add this to the base Trainable class.

Note also that this PR does NOT affect the behavior for users using the `setup_wandb()` utility and then log WandB metrics manually (for example in their Ray Train training functions).

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by justinvyu — APPROVED** [review]
Makes sense to me!

~~By the way, this will only fix the issue for the `setup_wandb` flow where users call that directly in their code. This does not solve the `WandbLoggerCallback` case -- should we fix it there too?~~

Nevermind. this is the exact line of code that was changed.

**Comment by sven1977** [comment]
Yeah, that's what I thought, too. Thanks for clarifying and for your review @justinvyu ! :)

File: python/ray/air/integrations/wandb.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 from ray._private.storage import _load_class
 from ray.air import session
 from ray.air._internal import usage as air_usage
+from ray.air.constants import TRAINING_ITERATION
```
Comment (simonsays1980) [inline python/ray/air/integrations/wandb.py:18]: If air goes anyways shouldn't this class maybe move over to `train` or `tune`?

File: python/ray/air/integrations/wandb.py [hunk 2]
```diff
@@ -415,7 +416,7 @@ def run(self):
             log, config_update = self._handle_result(item_content)
             try:
                 self._wandb.config.update(config_update, allow_val_change=True)
-                self._wandb.log(log)
+                self._wandb.log(log, step=log.get(TRAINING_ITERATION))
```
Comment (simonsays1980) [inline python/ray/air/integrations/wandb.py:419]: Sweet.

**Review by simonsays1980 — APPROVED** [review]
LGTM. Great improvement for users imo. Checkpointing is so important and should work as clean as possible for good user experience.

**Pull request merged**